deps/rabbit/src/rabbit_fifo.erl

```diff
@@ -299,7 +300,8 @@ apply(#{index := Index,
     Exists = maps:is_key(ConsumerId, Consumers),
     case messages_ready(State0) of
         0 ->
-            update_smallest_raft_index(Index, {dequeue, empty}, State0, []);
+            update_smallest_raft_index(Index, {dequeue, empty}, State0,
+                                       [notify_decorators_effect(State0)]);
```

Comment:
why are we notifying decorators here?

Reply:
There is a new consumer, so the federation can set the corresponding consumer on the upstream. Strictly it might only be needed when the first consumer appears and there is no other active consumer. I'll have a think if we can easily verify that